### PR TITLE
Add CodeVersioning page

### PR DESCRIPTION
**Pull request by horaklukas**
Add page about code versioning. For now there is only brief description of workflow models we use (for apps and for libs), but in future there could more topics related to version control.

### CODE REVIEW

File: pages/CodeVersioning.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Code versioning
+
+Every developer needs some [version control](https://en.wikipedia.org/wiki/Version_control) and we decided to use Git for that purpose. Here you can find some description of some best bractices or worflows we are trying keep when using it.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: ... workflows we are trying **to** keep...

File: pages/CodeVersioning.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Topics
+
+ - [Branching model](#branching-model)
+ 
+## Branching model
+
+There is several standard [git workflows](https://cs.atlassian.com/git/tutorials/comparing-workflows), and we use two of them: [Git flow](#git-flow-workflow) and [feature branch](#feature-branch-workflow) workflow. Which one we use depends on what we are developing.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: There are several...

File: pages/CodeVersioning.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### Git flow workflow
+
+That is a worflow we use when developing **applications**. There are always at least two persistent branches `master` and `develop` (`development`) as shown on the picture. Most time there is also `stage` branch that represents middle phase between `master` and `development`.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: ... as shown **in** the picture...

File: pages/CodeVersioning.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+All mentioned branches (`master`, `development`, `stage`) represents environment for deploying.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: Mozna spis “... for deployment.” 🤔

File: pages/CodeVersioning.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+### Git flow workflow
+
+That is a worflow we use when developing **applications**. There are always at least two persistent branches `master` and `develop` (`development`) as shown on the picture. Most time there is also `stage` branch that represents middle phase between `master` and `development`.
+
+All mentioned branches (`master`, `development`, `stage`) represents environment for deploying. 
+
+* `development` - we use it when developing application, every new feature or bug fix (not urgent) is first merged into development branch which is then deployed to development environment and tested if it works.
+* `stage` - when project manager decides that developed features, fixed bugs are ready to test by customer, they are usually first merged to `stage` branch and test at its environemnt before deploying to production
+* `master` - when new features, bug fixes are approved by testers (`development`) and customer (`stage`) they are ready to merge to `master` branch and develop to production environemnt.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: Nemyslel jsi tady “deploy to production...” misto develop?

File: pages/CodeVersioning.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+All mentioned branches (`master`, `development`, `stage`) represents environment for deploying. 
+
+* `development` - we use it when developing application, every new feature or bug fix (not urgent) is first merged into development branch which is then deployed to development environment and tested if it works.
+* `stage` - when project manager decides that developed features, fixed bugs are ready to test by customer, they are usually first merged to `stage` branch and test at its environemnt before deploying to production
+* `master` - when new features, bug fixes are approved by testers (`development`) and customer (`stage`) they are ready to merge to `master` branch and develop to production environemnt.  
+*Be aware that for `master` branch unlike for `development`, deploy is usually not started automatically and you need to start it manually.*
+
+![Git flow wofklow](https://buddy.works/blog/images/gitflow.png)
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: Typo “wofklow”

File: pages/CodeVersioning.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+*Image taken from https://buddy.works/blog/5-types-of-git-workflows*
+
+You can find more about the workflow at external resources below:
+
+* https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow
+* https://nvie.com/posts/a-successful-git-branching-model/
+
+### Feature branch workflow
+
+That is a worflow we use when developing **libraries** (see [tools we cook](./ToolsWeCook.md)). We found it more effecient than [Git flow](#git-flow-workflow) as there is usually no development version hold for some time and when new feature is finished, it's required to release immediately.
```
Comment (jstorm31) [inline pages/CodeVersioning.md]: Typo “worflow”

File: pages/CodeVersioning.md (hunk 1 above)
Comment (horaklukas) [reply pages/CodeVersioning.md]: 👍 vypadlo mi díky

File: pages/CodeVersioning.md (hunk 4 above)
Comment (horaklukas) [reply pages/CodeVersioning.md]: Máš pravdu, zní to líp.

File: pages/CodeVersioning.md (hunk 5 above)
Comment (horaklukas) [reply pages/CodeVersioning.md]: jj, of course 😄

File: pages/CodeVersioning.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+Every developer needs some [version control](https://en.wikipedia.org/wiki/Version_control) and we decided to use Git for that purpose. Here you can find some description of some best bractices or workflows we are trying to keep when using it.
+
+## Topics
+
+ - [Branching model](#branching-model)
+ 
+## Branching model
+
+There are several standard [git workflows](https://cs.atlassian.com/git/tutorials/comparing-workflows), and we use two of them: [Git flow](#git-flow-workflow) and [feature branch](#feature-branch-workflow) workflow. Which one we use depends on what we are developing.
```
Comment (baierjak) [inline pages/CodeVersioning.md]: standards?

Comment (baierjak) [reply pages/CodeVersioning.md]: tahle veta - Which one we use depends on what we are developing. je podle me nejak uplne blbe postavena jakoby byla cesky ... ne?

File: pages/CodeVersioning.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+### Git flow workflow
+
+That is a workflow we use when developing **applications**. There are always at least two persistent branches `master` and `develop` (`development`) as shown in the picture. Most time there is also `stage` branch that represents middle phase between `master` and `development`.
+
+All mentioned branches (`master`, `development`, `stage`) represents environment for deployment. 
+
+* `development` - we use it when developing application, every new feature or bug fix (not urgent) is first merged into development branch which is then deployed to development environment and tested if it works.
+* `stage` - when project manager decides that developed features, fixed bugs are ready to test by customer, they are usually first merged to `stage` branch and test at its environemnt before deploying to production
```
Comment (baierjak) [inline pages/CodeVersioning.md]: by a customer

Comment (baierjak) [reply pages/CodeVersioning.md]: od toho , they are usually first .... to je pak hrozne čingliš.... nebo tomu vubec nerozumim

**Comment by baierjak** [comment]
zkus trochu poupravit a pak mergni

File: pages/CodeVersioning.md (hunk 8 above)
Comment (horaklukas) [reply pages/CodeVersioning.md]: Standrd git workflows - to je správně.
Hele zrovna todle mi přijde vpohodě, rozhodně srozumitelnější než to co navrhuje Jirka

File: pages/CodeVersioning.md (hunk 9 above)
Comment (horaklukas) [reply pages/CodeVersioning.md]: Okej, co takhle: 

> when project manager decides that developed features or fixed bugs are ready to test by a customer, developer deploy them to the stage environment where customer can test them. That is accomplished by merging the `development` branch to the `stage` branch.

je to srozumitelnější?

File: pages/CodeVersioning.md (hunk 8 above)
Comment (jstorm31) [reply pages/CodeVersioning.md]: Jo, asi v pohodě. Můžeme se pak zeptat Majkla, jak by to napsal 😄

**Pull request merged**